src/server.rs

```diff
@@ -973,33 +975,43 @@ impl LanguageServer for LspServer {
         };
         Ok(Some(response))
     }
-    async fn references(
+    async fn document_highlight(
         &self,
-        params: lsp::ReferenceParams,
-    ) -> RpcResult<Option<Vec<lsp::Location>>> {
+        params: lsp::DocumentHighlightParams,
+    ) -> RpcResult<Option<Vec<lsp::DocumentHighlight>>> {
         let key =
-            params.text_document_position.text_document.uri.clone();
-        let store = match self.store.lock() {
-            Ok(value) => value,
+            params.text_document_position_params.text_document.uri;
+        let contents = self.get_document(&key)?;
```

Comment:
Nit: `get_document` into `parse_and_analyze` is called a lot, should factor it out into an helper.

Reply:
Refactored, a nice side effect is that now error handling is consistent for these cases.